synapse/handlers/message.py

```diff
@@ -1432,18 +1432,10 @@ async def _persist_events(
         PartialStateConflictError: if attempting to persist a partial state event in
             a room that has been un-partial stated.
         """
-
-        for event, context in events_and_context:
-            # Skip push notification actions for historical messages
-            # because we don't want to notify people about old history back in time.
-            # The historical messages also do not have the proper `context.current_state_ids`
-            # and `state_groups` because they have `prev_events` that aren't persisted yet
-            # (historical messages persisted in reverse-chronological order).
-            if not event.internal_metadata.is_historical():
-                with opentracing.start_active_span("calculate_push_actions"):
-                    await self._bulk_push_rule_evaluator.action_for_event_by_user(
-                        event, context
-                    )
+        with opentracing.start_active_span("calculate_push_actions"):
```

Comment:
I think we're very much changing the meaning of this space by moving it "out" of the loop over events.

I wonder if we want to move this into `action_for_events_by_user` and wrap just the call to `_action_for_event_by_user`. But that's currently decorated with `measure_func`...why would we trace those separately!? 😕

Reply:
I added the opentracing span awhile back as part of the local join profiling, and I think you're right that this information is captured by the `measure_func` and we could probably ditch the opentracing span.

Reply:
I don't feel strongly, but if we don't find it useful than ditching is probably for the best.

Right now it will give inaccurate reports if you look at it over a release before and after this change (e.g. if we persist 100 events at once, we've now 100xed the meaning of this value).

I think it essentially measures the same thing as the `measure_func` call, which *I think* we want to keep on `_action_for_event_by_user` so that it still measures a *per event* recording.